src/modules/menu/components/menu/common/EmailValidationField.vue

```diff
@@ -0,0 +1,45 @@
+<script setup>
+import { computed, ref, watch } from 'vue'
+
+import { isValidEmail } from './utils'
+
+const emits = defineEmits(['email-updated'])
+const props = defineProps({
+    isRequestPending: {
+        type: Boolean,
+        default: false,
+    },
+})
+
+// Reactive data
+const email = ref('')
+const userIsTypingEmail = ref(false)
+
+// Computed properties
+const isEmailValid = computed(() => {
+    return !email.value || isValidEmail(email.value)
+})
+// Watches
+watch(email, () => {
+    emits('email-updated', email.value)
+})
+</script>
+
+<template>
+    <div>
+        <span><slot name="header"></slot></span>
```

Comment:
Why using a slot for the field label ? I think it would be better to use a props containing a translation key to use. The label should also use `<label></label>` tag using the `for` attribute. See bootstrap examples (https://getbootstrap.com/docs/4.0/components/forms/).
Instead of using `isRequestPending` props which is quite specific I would use `disabled` props, letting the user defining what event/action is disabling the component.
Finally this component should be moved to src/utils/components/ as it is very generic and could be used any where in the application.